test/e2e/framework.go

```diff
@@ -2718,30 +2705,46 @@ func (data *TestData) gracefulExitFlowAggregator(covDir string) error {
 	}
 	podName := flowAggPod.Name
 
-	if err := data.killProcessAndCollectCovFiles(flowAggregatorNamespace, podName, "flow-aggregator", flowAggregatorCovBinary, flowAggregatorCovFile, covDir); err != nil {
+	if err := data.killProcessAndCollectCovFiles(flowAggregatorNamespace, podName, "flow-aggregator", flowAggregatorCovBinary, covDir); err != nil {
 		return fmt.Errorf("error when gracefully exiting Flow Aggregator: %w", err)
 	}
 
 	return nil
 }
 
-// collectAntctlCovFiles collects coverage files for the antctl binary from the Pod and saves them to the coverage directory
-func (data *TestData) collectAntctlCovFiles(podName string, containerName string, nsName string, covDir string) error {
+// collectCovFiles collects coverage files from the Pod and saves them to the coverage directory
+func (data *TestData) collectCovFiles(podName string, containerName string, nsName string, covDir string) error {
 	// copy antctl coverage files from Pod to the coverage directory
-	cmds := []string{"bash", "-c", "find . -maxdepth 1 -name 'antctl*.out' -exec basename {} ';'"}
+	cmds := []string{"bash", "-c", "find /tmp/coverage  -mindepth 1"}
 	stdout, stderr, err := data.RunCommandFromPod(nsName, podName, containerName, cmds)
 	if err != nil {
 		return fmt.Errorf("error when running this find command '%s' on Pod '%s', stderr: <%v>, err: <%v>", cmds, podName, stderr, err)
 	}
+	covDir = filepath.Join(covDir, podName+"-coverage")
+	for i := 0; true; i++ {
```

Comment:
this is strange, do we expect the loop to execute multiple times or in general will it "succeed" at the first iteration (`i = 0`)? maybe we could have a comment

Reply:
My initial thinking while adding this was: if the same agent pod is used to run multiple tests, and generate coverage files multiple times, they should be separated. So I made sure that they are stored in different directories. BUT, as far as I have locally tested, only one directory is needed for one pod. So now I think this can be removed.

Reply:
My understanding is that we only collect the coverage files when we terminate the Agent / Controller Pod, so there can only be one collection event for ant given Pod.